aws-api/src/androidTest/java/com/amplifyframework/api/aws/SubscriptionEndpointTest.java

```diff
@@ -0,0 +1,156 @@
+    @After
+    public void releaseSubscriptions() throws RuntimeException {
+        for (String subscriptionId : subscriptionIdsForRelease) {
+            executor.execute(() ->
+                Await.result((onResult, onError) -> {
+                    try {
+                        subscriptionEndpoint.releaseSubscription(subscriptionId);
+                    } catch (ApiException failureToLetGo) {
+                        // Friend, it is time. It is time.
+                        throw new RuntimeException(failureToLetGo);
+                    }
+                })
+            );
+        }
+    }
+
+    /**
+     * It should be possible to create two subscriptions to the same type of model.
+     * @throws ApiException On failure to subscribe
+     */
+    @Test
+    public void twoSubscriptionsToTheSameThing() throws ApiException {
+        // Okay, request a first subscription.
+        String firstSubscriptionId = subscribeToEventComments(eventId);
+        assertNotNull(firstSubscriptionId);
+        subscriptionIdsForRelease.add(firstSubscriptionId);
+
+        // Now, request a second subscription, with the same request data.
+        // This is around where the test fails.
+        String secondSubscriptionId = subscribeToEventComments(eventId);
+        assertNotNull(secondSubscriptionId);
+        subscriptionIdsForRelease.add(secondSubscriptionId);
+
+        // Theoretically would expect that we'd have two subscriptions,
+        // and that their subscription IDs would be different/unique.
+        assertNotEquals(firstSubscriptionId, secondSubscriptionId);
+    }
+
+    /**
+     * Subscribe to comments on an event.
+     * @param eventId ID of event for which comments are being made
+     * @return Subscription ID received from subscription_ack message payload
+     * @throws ApiException If outcome of subscription request is anything other than an ACK w/ new ID
+     */
+    private String subscribeToEventComments(String eventId) throws ApiException {
+        // Arrange a request to start a subscription.
+        String document = Assets.readAsString("subscribe-event-comments.graphql");
+
+        GsonVariablesSerializer serializer = new GsonVariablesSerializer();
+        Map<String, Object> variables = Collections.singletonMap("eventId", eventId);
+        GraphQLRequest<String> request = new GraphQLRequest<>(document, variables, String.class, serializer);
+
+        return Await.<String, ApiException>result((onResult, onError) ->
+            executor.execute(() ->
+                subscriptionEndpoint.requestSubscription(
+                    request,
+                    onResult,
+                    item -> {
+                        final String message;
+                        if (item.hasErrors()) {
+                            message = "Subscription error: " + item.getErrors().toString();
+                        } else {
+                            message = "Unexpect subscription data: " + item.getData();
```

Comment:
*Unexpected

Reply:
Also, I must be missing something but why are you failing if it receives an item without errors?

Reply:
Ah thanks the for the catch -

And, for _this_ test, all that's important is the establishment of the subscription, and the receipt of the subscription IDs. Since I don't create any actual models, I don't expect to receive any subscription data. If I do receive any items on the subscription, it'd be unexpected.